src/modules/job-manager/prioritize.c

```diff
@@ -114,6 +115,9 @@ static int reprioritize_one (struct job_manager *ctx,
                              "priority", priority) < 0)
         return -1;
 
+    // update priority
+    job->priority = priority;
+
```

Comment:
Ah, I wonder if the behavior changed here when we stopped recursively posting events. Note the comment above says:

```c
   /* Post 'priority' event.
     *
     *  This call will result in job->priority being set, and,
     *   if the job is in the PRIORITY state, will transition to
     *   the SCHED state, invoke plugin callbacks, etc.
     */
```

This may no longer be the case since the event could be placed in the event queue instead of being posted immediately.

Note that the way this is supposed to work is that when the event handling mechanism in `job-manager/event.c` actually _handles_ the `priority` event, that should set `job->priority`. I wonder if the correct fix is to move all the other stuff in this function after `event_job_post_pack()` to the action for the `priority` event? (Though this approach is much simpler)

I think I'd need @garlick's input on that.

Reply:
Oops, sorry, I cross posted with @garlick.

Reply:
NP!  _seems_ like it wouldn't matter where that happens but I can't recall all the details of how the accounting plugin uses this so it might be worth reviewing the RFC and the code a bit first.

I'm on vaca today and likely won't have time but I'm sure I'll be fine with whatever is decided if this need to be fixed now.  O/W I'll check back later this weekend.

Nice job running that down @trws.

Reply:
We'll need a fix before the async queues PR can land, but I think the plan was early next week anyway.  Normally I'd be happy to move the code over to the event handling, but honestly I'm not sure where that's happening, even with the pointer to event.c.  If one of you can point me in the right direction I can certainly do that though.

Reply:
The priority event is handled here I think

https://github.com/flux-framework/flux-core/blob/da43fd3290744ea865d98ba43611c3c57634dbd8/src/modules/job-manager/event.c#L619-L627

But it may require more thought to determine if it is correct to move that code there (and I've got to sign off soon). It does _seem_ correct that the actual _handling_ of the `priority` event should trigger the queue reprioritization and update of the scheduler etc. However, when all jobs are reprioritized, I don't think we want to take this approach (which may re-sort the queue once for every affected job), which is probably why the code is the way it is now...

Reply:
My honest inclination would be to leave this fix as-is with a big comment explaining what probably _should_ happen (and open a corresponding issue to look into it). I _think_ it is safe here because that is how the code was originally intended to work.

This is because the async queues PR is very high priority right now.

However, I'd want to get an ACK from @garlick first when he's back, and perhaps @cmoussa1 as well just in case I've missed something obvious.

Reply:
Sounds right to me.  The concern I would have is that we _might_ see "bouncing" if multiple reprio requests are fired for the same job in quick succession and the value is being set in two places.  Could resolve that by removing the other set of the value from the event and test the result, but I'm not sure if we want to do that or not.